modules/Stories/Stories.tsx

```diff
@@ -1,27 +1,46 @@
+import type { Category } from '@prezly/sdk';
 import type { Locale } from '@prezly/theme-kit-nextjs';
 
 import { app } from '@/adapters/server';
 
 import { InfiniteStories } from '../InfiniteStories';
 
 interface Props {
+    categoryId: Category['id'] | undefined;
     localeCode: Locale.Code;
     pageSize: number;
 }
 
-export async function Stories({ localeCode, pageSize }: Props) {
+export async function Stories({ categoryId, localeCode, pageSize }: Props) {
     const newsroom = await app().newsroom();
     const languageSettings = await app().languageOrDefault(localeCode);
+    const categories = await app().categories();
+    const featuredCategories = categories.filter(
+        ({ is_featured, i18n }) => is_featured && i18n[localeCode]?.public_stories_number > 0,
+    );
+    const hasFeaturedCategories = featuredCategories.length > 0;
+
+    const { stories: pinnedOrMostRecentStories } = await app().stories({
+        limit: 1,
+        locale: { code: localeCode },
+    });
+
     const { stories, pagination } = await app().stories({
+        category: categoryId ? { id: categoryId } : undefined,
         limit: pageSize,
         locale: { code: localeCode },
     });
 
     return (
         <InfiniteStories
+            key={categoryId}
```

Comment:
I've added the key prop here so the component is re-mounted and the `useInfiniteLoading` hook resets its state since we're providing different `initialStories`.